### PR TITLE
👽 Fix breaking changes in react native API

### PR DESCRIPTION
**Pull request by williamchong**

### CODE REVIEW

File: src/screensComponents/AuthcoreScreen.jsx [hunk 1]
```diff
@@ -55,14 +55,19 @@ export default class AuthcoreScreen extends Component {
...
+        window.addEventListener('load', () => { window.postMessage({ type: 'AuthCore_accessToken', data: '${accessToken}'}, '${this.client.baseUrl}') })
+      }
     `
 
     return (
       <WebView
         containerStyle={ containerStyle }
         source={{ uri: this.widgetPath }}
         injectedJavaScript={accessToken !== undefined && injectAccessToken}
+        onMessage={() => {}}
```
Comment (nwingt) [inline src/screensComponents/AuthcoreScreen.jsx:70]: What is this for?

File: src/webauth/agent.js [hunk 2]
```diff
@@ -15,15 +19,21 @@ export default class Agent {
...
+      this.urlEventListener = Linking.addEventListener('url', urlHandler)
       if (!triggerError) {
         Authcore.showNormalUrl(url, closeOnLoad)
       } else {
         Authcore.showAuthUrl(url, closeOnLoad, (error, redirectURL) => {
-          Linking.removeEventListener('url', urlHandler)
+          if (this.urlEventListener) {
+            this.urlEventListener.remove()
+            this.urlEventListener = null
+          }
```
Comment (nwingt) [inline src/webauth/agent.js:36]: If `Linking` is removed, should remove import too?

File: src/screensComponents/AuthcoreScreen.jsx (hunk 1 above)
Comment (williamchong) [reply src/screensComponents/AuthcoreScreen.jsx:70]: needed for `injectedJavaScript` to work on iOS
https://github.com/react-native-webview/react-native-webview/blob/master/docs/Reference.md#injectedjavascript

File: src/webauth/agent.js (hunk 2 above)
Comment (williamchong) [reply src/webauth/agent.js:36]: We need it for `Linking.addEventListener`

**Review by nwingt — APPROVED** [review]

**Pull request merged**